[AIRFLOW-3758] Fix circular import in WasbTaskHandler

ttanay opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3758
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

Import for WasbHook in airflow.utils.log.WasbTaskHandler was causing a circular import when configure_logging() was called.
Closes [AIRFLOW-3758].

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
   - Test configure_logging() with the settings for logging to Azure Blob Storage.
### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

ttanay committed:
[AIRFLOW-3758] Defer WasbHook import

WasbHook was causing a circular import when configure_logging() was called.

[AIRFLOW-3758] Add test for writing logs to Azure Blob

configure_logging() is called with the settings to write to Azure
Blob Storage. If the log event is as expected, the test passes.

ashb commented on tests/test_logging_config.py:
```diff
@@ -95,6 +95,55 @@
 """
 
+SETTINGS_FILE_WASB_HANDLER = """
+REMOTE_BASE_LOG_FOLDER = 'wasb'
```
Would it be possible to test this with the _default_ logging config  and the core__remote_logging settings instead?

ashb replied:
Something like this might help

```python
from six.moves import reload_module

def x():
   import airflow.config_templates.airflow_local_settings
   reload_module(airflow.config_templates.airflow_local_settings)
```

ttanay replied:
> Would it be possible to test this with the _default_ logging config and the core__remote_logging settings instead?

I dug through the code for loading the log configs for remote logging. 
Looks like its as simple as changing the following settings:
1. `core__remote_logging`
2. `core__remote_log_conn_id`
3. `core__remote_base_log_folder`

This is in line with the changes that need to be made in order to [write logs to S3 or GCS.](https://airflow.apache.org/howto/write-logs.html#writing-logs-to-amazon-s3)
The docs describe a longer process for [writing logs to Azure Blob Storage](https://airflow.apache.org/howto/write-logs.html#writing-logs-to-azure-blob-storage). 
The docs also need to be updated. Will update them in a separate [AIRFLOW-XXX] PR.
Updating the test.

Thanks @ashb

ttanay committed:
Don't load custom logging_config_class in test

To send logs to an Azure Blob, a custom logging_config_class may
not be specified. Just changing the configs for remote_logging
will work.

ashb commented on tests/test_logging_config.py:
```diff
@@ -251,6 +251,14 @@ def test_1_9_config(self):
+        conf.set('core', 'remote_logging', 'True')
+        conf.set('core', 'remote_log_conn_id', 'some_wasb')
+        conf.set('core', 'remote_base_log_folder', 'wasb://some-folder')
+        configure_logging()
```
I know we were just testing that it loaded without error, but it always feels a bit odd to not see any assertions in a test.

How about we add something like

```python
import airflow.utils.log.wasb_task_handler.WasbTaskHandler
logger = logging.getLogger('airflow.task.x')
self.assertIsInstance(logger.handlers[0], airflow.utils.log.wasb_task_handler.WasbTaskHandler)
```

ashb replied:
(I've done that from memory- that may not be the property name on logger)

ttanay replied:
Sure. I'll add that.

ashb replied:
@ttanay Any chance you'd have a chance to look at this last change?

ttanay replied:
Hey @ashb 
Didn't get a chance to look at it. Exams :/
Will get it done by Tuesday.

ttanay replied:
Hey @ashb 
I tried this after `configure_logging` is called in the change above, and also in a DAG:
```
import logging
logger = logging.getLogger('airflow.task.x')
```
But, the `logger.handlers` was an empty list `[]`.

Am I missing something?

ashb replied:
Hmmm, I'll double check. Did you also try with just `logging.getLogger('airflow.task')`

ttanay replied:
No, but, it works now. Thanks!

ashb replied:
Ah that makes sense. `airflow.task.x` isn't directly configured with handlers (hence the empty list) but when emitting messages the logging framework will go up the hierarchy until it finds some (that's a little bit simplified, but close)

ttanay replied:
Oh. Now I get it. Thanks!

ttanay committed:
Add extra assert to verify if handler was added

ashb commented on tests/test_logging_config.py:
```diff
@@ -259,6 +259,11 @@ def test_loading_remote_logging_with_wasb_handler(self):
+        import logging
+        from airflow.utils.log.wasb_task_handler import WasbTaskHandler
+        logger = logging.getLogger('airflow.task')
+        self.assertIsInstance(logger.handlers[0], WasbTaskHandler)
```
Failing with:

`   AssertionError: <airflow.utils.log.file_task_handler.FileTaskHandler object at 0x7f5bf7d1a950> is not an instance of <class 'airflow.utils.log.wasb_task_handler.WasbTaskHandler'>`

Because of how configure logging works we will need to reload  before calling configure_ logging() airflow.config_templates.airflow_local_settings:

```
from six.moves import reload_module
import airflow.config_templates.airflow_local_settings
reload_module(airflow.config_templates.airflow_local_settings)
configure_logging()
```

ttanay committed:
Reload settings before configuring logging
